projects/ngneat/reactive-forms/src/lib/formGroup.ts

```diff
@@ -48,56 +57,67 @@ export class FormGroup<T extends Obj = any, E extends object = any> extends NgFo
   touch$ = this.touchChanges.asObservable().pipe(distinctUntilChanged());
   dirty$ = this.dirtyChanges.asObservable().pipe(distinctUntilChanged());
 
-  readonly value$ = controlValueChanges$<T>(this);
-  readonly disabled$ = controlDisabled$<T>(this);
-  readonly enabled$ = controlEnabled$<T>(this);
-  readonly status$ = controlStatusChanges$<T>(this);
+  readonly value$ = controlValueChanges$<ControlsValue<T>>(this);
+  readonly disabled$ = controlDisabled$<ControlsValue<T>>(this);
+  readonly enabled$ = controlEnabled$<ControlsValue<T>>(this);
+  readonly status$ = controlStatusChanges$<ControlsValue<T>>(this);
   readonly errors$ = controlErrorChanges$<E>(this);
 
-  constructor(
-    public controls: ExtractAbstractControl<KeyValueControls<T>, T>,
-    validatorOrOpts?: ValidatorOrOpts,
-    asyncValidator?: AsyncValidator
-  ) {
+  constructor(public controls: ControlsOfValue<T>, validatorOrOpts?: ValidatorOrOpts, asyncValidator?: AsyncValidator) {
```

Comment:
will it still be typed for nested forms that includes `FormGroup` or `FormArray` without passing the `FormGroup` or `FormArray` types? 
for example:
```ts
interface ProfileControls {
  firstName: string;
  lastName: string;
  address: {
    street: string,
    city: string
  } 
}

const profileForm = new FormGroup<ProfileControls>({
  firstName: new FormControl(''),
  lastName: new FormControl(''),
  address : new FormGroup({
    street: new FormControl(''),
    city  : new FormControl('')
  })
});
```

will city and still be typed?

Reply:
Yes, in that case the form controls will be interpreted as:
```ts
{
  firstName: FormControl<string>;
  lastName: FormControl<string>;
  address: AbstractControl<{
    street: string;
    city: string;
  }>
}
```

Which means that:
```ts
profileForm.getRawValue().address.city; // string
profileForm.getControl('address'); // AbstractControl<{street: string; city: string}>
profileForm.getControl('address', 'street'); // AbstractControl<string>
```
Exactly as it was before these changes.